EDSC-4031: Always route to saved Projects when selected.

Bccorb opened the pull request:
# Overview

### What is the feature?

Created a SavedProjects page and route to avoid the accidental appending of route params to the project page.

### What is the Solution?

- New Component
- New Route
- Updated the link for saved projects to the new route

### What areas of the application does this impact?

- Saved Projects
- Projects when missing all params

# Testing

### Reproduction steps

This is confirmable by:

1. Logging in to EDSC
2. Selecting the save project icon in the right hand tool bar
3. Entering a project name
4. Adding a collection to the project via the plus button on a collection
5. Clicking the user icon in the right hand tool bar and selecting "saved projects"
6. You should observe that you are routed to  '/savedProjects' and it is in face your saved projects.

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

eudoroolivares2016 commented on static/src/js/routes/SavedProjects/SavedProjects.js:
```diff
@@ -0,0 +1,40 @@
+import SavedProjectsContainer from '../../containers/SavedProjectsContainer/SavedProjectsContainer'
+import AuthRequiredContainer from '../../containers/AuthRequiredContainer/AuthRequiredContainer'
+
+export class SavedProjects extends Component {
```
We should make this a functional component like in `static/src/js/routes/Preferences/Preferences.js`

eudoroolivares2016 commented on static/src/js/routes/SavedProjects/__tests__/SavedProjects.test.js:
```diff
@@ -0,0 +1,60 @@
+  createControlComponent: jest.fn().mockImplementation(() => {})
+}))
+
+Enzyme.configure({ adapter: new Adapter() })
```
If we can avoid it we should use RTL on new tests so we don't accumulate new tech debt

Bccorb replied:
Great point! I converted all of these! Thank you for keeping me honest!

Bccorb commented:
> `/savedProjects` is not a valid URL, it should be `/saved-projects` to match every other URL in `App.js`.
> 
> Instead of fixing the bug in the ticket you added a new feature and left the original buggy code abandoned in the codebase. The `Project.js` route already has code with comments that explain how the behavior should work. This is occurring because `?projectId=...` is still in the URL when it switches to the `/projects` endpoint. Either track down the bug and fix it, or remove the buggy code.
> 
> See CONTRIBUTING.md, _all_ commits have to start with the ticket number.

All feedback incorporated. 
- Removed saved project logic from the project url
- Added logic to correct an issue where a project url with no params could still be navigated to
- Added the back to search to the saved-projects url
- renamed the url endpoint

eudoroolivares2016 commented on static/src/js/routes/Project/Project.js:
```diff
@@ -66,24 +65,10 @@ export class Project extends Component {
 
-    // If there are no params in the URL, show the saved projects page
+    // If there are no project params in the URL, re-route to the saved Projects page
     if (search === '') {
```
I'm a bit confused I'll pull this down and test it myself but, if the conditional here `if (search === '')` is failing to properly work i.e. its `false` when it should be true, which is why we are getting routed to the other component wouldn't this still have the same issue even if  it is on a difference page?

eudoroolivares2016 replied:
So the issue I seem to be seeing is that something like `http://localhost:8080/saved-projects?projectId=3519105921`is always going to strip out the projectId but, also that `http://localhost:8080/projects?projectId=3519105921` is redirecting me (fill in any projectId) back into the `saved-projects` page so I can't go back to my previous projects

eudoroolivares2016 commented on static/src/js/routes/SavedProjects/SavedProjects.js:
```diff
@@ -0,0 +1,31 @@
+        <title>Saved Projects</title>
+        <meta name="title" content="Saved Projects" />
+        <meta name="robots" content="noindex, nofollow" />
+        <link rel="canonical" href={`${edscHost}/projects`} />
```
```suggestion
        <link rel="canonical" href={`${edscHost}/saved-projects`} />
```

eudoroolivares2016 commented:
Update I was looking at this some because of a related ticket I have. It seems like the issue is that we can't really differentiate between an empty project and the saved project page. Its possible to fix the issue where it will go to saved project. The way that we are currently checking makes sense i.e. if there are search parameters then we go to the project page the issue is that somehow the saved projects page is carrying some search parameters when clicking on the saved projects page

eudoroolivares2016 commented:
Closing PR likely will use a different solution otherwise we'll reference this one but, its gotten quite behind the `HEAD`
